Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/VirtualReality.java

```diff
@@ -0,0 +1,150 @@
+{
+  private static final String VR_PACKAGE = "org.dolphinemu.dolphinemu.vr";
+
+  public static boolean isActive()
+  {
+    return BuildConfig.BUILD_TYPE.equals("vr");
+  }
+
+  public static boolean isInstalled(Context context)
+  {
+    PackageManager pm = context.getPackageManager();
+    for (ApplicationInfo app : pm.getInstalledApplications(PackageManager.GET_META_DATA))
+    {
+      if (app.packageName.equals(VR_PACKAGE))
+      {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  public static boolean isLegacyPath(String filePath)
+  {
+    return filePath.startsWith("/");
+  }
+
+  public static void linkLoader()
+  {
+    if (isActive())
+    {
+      String manufacturer = Build.MANUFACTURER.toLowerCase(Locale.ROOT);
+      // rename oculus to meta as this will probably happen in the future anyway
+      if (manufacturer.contains("oculus"))
+        manufacturer = "meta";
+
+      //Load manufacturer specific loader
+      try
+      {
+        System.loadLibrary("openxr_" + manufacturer);
+      }
+      catch (UnsatisfiedLinkError e)
+      {
+        Log.error("Unsupported VR device: " + manufacturer);
+        System.exit(0);
+      }
+    }
+  }
+
+  public static void openIntent(Context context, String[] filePaths)
+  {
+    Intent launcher = context.getPackageManager().getLaunchIntentForPackage(VR_PACKAGE);
+
+    if (VirtualReality.isLegacyPath(filePaths[0]))
+    {
+      launcher.putExtra("AutoStartFiles", filePaths);
+    }
+    else
+    {
+      Uri uri;
+      try
+      {
+        uri = ContentHandler.unmangle(filePaths[0]);
+      }
+      catch (FileNotFoundException e)
+      {
+        uri = Uri.parse(filePaths[0]);
+      }
+      launcher.addFlags(Intent.FLAG_GRANT_PERSISTABLE_URI_PERMISSION);
+      launcher.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+      launcher.setAction(Intent.ACTION_GET_CONTENT);
+      launcher.setData(uri);
+    }
+
+    serializeConfigs(launcher);
+    context.getApplicationContext().startActivity(launcher);
+  }
+
+  public static void restoreConfig(Context context, Bundle extras)
+  {
+    File root = DirectoryInitialization.getUserDirectoryPath(context);
+    File config = new File(root, "/Config/");
+    config.mkdirs();
+
+    for (String filename : extras.getStringArrayList("ConfigFiles"))
+    {
+      File file = new File(config, filename);
+
+      try
+      {
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O)
+        {
+          byte[] bytes = extras.getByteArray(file.getName());
+          if (bytes != null)
+            Files.write(file.toPath(), bytes);
+        }
+      }
+      catch (IOException e)
+      {
+        e.printStackTrace();
+      }
+    }
+  }
+
+  private static void serializeConfigs(Intent intent)
+  {
+    ArrayList<String> files = new ArrayList<>();
+    File root = new File(DirectoryInitialization.getUserDirectory() + "/Config/");
+    for (File file : root.listFiles())
```

Comment:
This is very brittle. You're going to miss settings that aren't in Config, like the SYSCONF file, which is in the Wii directory. Speaking of the Wii directory, is there a reason why you want to sync the config but not the saves?

Reply:
This is a temporary solution to transfer the basic settings from Android to VR APK storage. Syncing the save games from the VR APK storage into the Android APK storage would add unnecessary complexity.

Once there is the new GUI for VR build, there will be no need to have two APKs, so we can drop this code.